packages/edit-site/src/components/sidebar-navigation-screen-details-footer/index.js

```diff
@@ -14,32 +17,83 @@ import {
 	SidebarNavigationScreenDetailsPanelValue,
 } from '../sidebar-navigation-screen-details-panel';
 
+function ScreenDetailsLastModifiedDate( { lastModifiedDateTime } ) {
+	return (
+		<SidebarNavigationScreenDetailsPanelRow
+			className="edit-site-sidebar-navigation-screen-details-footer__revision"
+			justify="space-between"
+		>
+			<SidebarNavigationScreenDetailsPanelLabel>
+				{ __( 'Last modified' ) }
+			</SidebarNavigationScreenDetailsPanelLabel>
+			<SidebarNavigationScreenDetailsPanelValue>
+				{ createInterpolateElement(
+					sprintf(
+						/* translators: %s: is the relative time when the post was last modified. */
+						__( '<time>%s</time>' ),
+						humanTimeDiff( lastModifiedDateTime )
+					),
+					{
+						time: <time dateTime={ lastModifiedDateTime } />,
+					}
+				) }
+			</SidebarNavigationScreenDetailsPanelValue>
+			<Icon
+				className="edit-site-sidebar-navigation-screen-details-footer__icon"
+				icon={ backup }
+			/>
+		</SidebarNavigationScreenDetailsPanelRow>
+	);
+}
 export default function SidebarNavigationScreenDetailsFooter( {
-	lastModifiedDateTime,
+	record,
+	onClickRevisions,
 } ) {
+	/*
+	 * There might be other items in the future,
+	 * but for now it's just modified date,
+	 * so return null if there's no record?.modified.
+	 */
+	if ( ! record?.modified ) {
+		return null;
+	}
+
+	const lastRevisionId = record?._links?.[ 'predecessor-version' ]?.[ 0 ]?.id;
+
+	let revisionsButtonNavigationProps = null;
+
+	if ( onClickRevisions ) {
+		revisionsButtonNavigationProps = {
+			onClick: onClickRevisions,
+		};
+	}
+
+	if ( lastRevisionId ) {
+		revisionsButtonNavigationProps = {
+			href: addQueryArgs( 'revision.php', {
+				revision: lastRevisionId,
+				gutenberg: true,
+			} ),
+		};
+	}
+
 	return (
-		<>
-			{ lastModifiedDateTime && (
-				<SidebarNavigationScreenDetailsPanelRow className="edit-site-sidebar-navigation-screen-details-footer">
-					<SidebarNavigationScreenDetailsPanelLabel>
-						{ __( 'Last modified' ) }
-					</SidebarNavigationScreenDetailsPanelLabel>
-					<SidebarNavigationScreenDetailsPanelValue>
-						{ createInterpolateElement(
-							sprintf(
-								/* translators: %s: is the relative time when the post was last modified. */
-								__( '<time>%s</time>' ),
-								humanTimeDiff( lastModifiedDateTime )
-							),
-							{
-								time: (
-									<time dateTime={ lastModifiedDateTime } />
-								),
-							}
-						) }
-					</SidebarNavigationScreenDetailsPanelValue>
-				</SidebarNavigationScreenDetailsPanelRow>
+		<div className="edit-site-sidebar-navigation-screen-details-footer">
+			{ revisionsButtonNavigationProps ? (
+				<Button
```

Comment:
In case it relates to the issue of text alignment for the button, prior to this PR the Styles Details footer used a `SidebarNavigationItem` component. Internally, that winds up using an `Item` component. When passed in an `onClick` event, that component gets switched to use a button element [here](https://github.com/WordPress/gutenberg/blob/2fe728e4e12040a592804af76a57b6fc16eb8004/packages/components/src/item-group/item/hook.ts#L38), and overrides for text alignment occur [here](https://github.com/WordPress/gutenberg/blob/da4d91f8e0077910252c7c0feb3c5e67ec335d6d/packages/components/src/item-group/styles.ts#L19), so re-using `SidebarNavigationItem` might get us the desired overrides for free without having to add more in this PR?

Reply:
Thanks for that. I did try `SidebarNavigationItem` first but ran into some alignment woes when there was no link. I didn't exhaust all avenues so it worth a shot again to remove the logic from this component. 👍🏻